apps/token/src/routes/governance/components/proposal-detail-header/proposal-header.tsx

```diff
@@ -0,0 +1,104 @@
+import { useTranslation } from 'react-i18next';
+import { Lozenge } from '@vegaprotocol/ui-toolkit';
+import type { ReactNode } from 'react';
+import type { Proposals_proposals } from '../../proposals/__generated__/Proposals';
+
+export const ProposalHeader = ({
+  proposal,
+}: {
+  proposal: Proposals_proposals;
+}) => {
+  const { t } = useTranslation();
+  const { change } = proposal.terms;
+
+  let headerText: string;
+  let detailsOne: ReactNode;
```

Comment:
suggestion (non blocking): Perhaps just have a single `content` variable here set to null initially. This will save rendering out empty divs at the end as detailsTwo is only used for freeform proposals

Reply:
Ah good spot, thanks. I think I'll conditionally render detailsTwo as having the content and elements split like that aids testing.